module/Application/config/module.config.php

```diff
@@ -126,6 +127,11 @@
             'bootstrapElementError' => function () {
                 return new BootstrapElementError();
             },
+            'companyIdentity' => function (ContainerInterface $container) {
```

Comment:
I also propose not creating a seperate helper for this. Instead, it may be a better idea to introduce an interface for the Identity which both the UserModel and the CompanyUserModel implement. Then you could just reuse the same view helper.

Reply:
The problem is that the normal `identity` view helper only accepts a single `AuthenticationService` and our logic very clearly separates the two identities we have because they use different underlying models (`User` vs `CompanyUser`) and there is no way to separate the two without us having to store more information in our sessions.

Reworking the `AuthentiationService` to be one thing for both identities or changing our storage logic (how would you deal with the storage-level separation we currently have?) does not outweigh the benefits of not having a separate `companyIdentity()` view helper.

Additionally, we would have to change all our invocations of `identity()` to make sure we are working with a `User` and not a `CompanyUser`.

Reply:
We could change the names, make `userIdentity()` an alias of `identity()` and change `companyIdentity()` to `companyUserIdentity()`. Just like the `get...` methods you introduced.

Reply:
One way to make this easier, would be to have a superclass for User and CompanyUser. Then you could have one mapper for this. The main difficulty with this is that UserModels have a lidnr, and CompanyUserModels do not.

Reply:
Then I'd say that it is out of scope for this PR. There are many things that are more important and should be done instead.

If we make it work in the future that be great, although all the interdependencies may undo what you are trying to accomplish.